// devextreme-angular/esm2022/core/tokens/index.mjs
/*!
 * devextreme-angular
 * Version: 25.2.3
 * Build date: Fri Dec 12 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */
import { InjectionToken } from '@angular/core';
export const PROPERTY_TOKEN_items = new InjectionToken('property-token-items');
export const PROPERTY_TOKEN_buttons = new InjectionToken('property-token-buttons');
export const PROPERTY_TOKEN_toolbarItems = new InjectionToken('property-token-toolbarItems');
export const PROPERTY_TOKEN_validationRules = new InjectionToken('property-token-validationRules');
export const PROPERTY_TOKEN_changes = new InjectionToken('property-token-changes');
export const PROPERTY_TOKEN_columns = new InjectionToken('property-token-columns');
export const PROPERTY_TOKEN_customOperations = new InjectionToken('property-token-customOperations');
export const PROPERTY_TOKEN_fields = new InjectionToken('property-token-fields');
export const PROPERTY_TOKEN_tabs = new InjectionToken('property-token-tabs');
export const PROPERTY_TOKEN_annotations = new InjectionToken('property-token-annotations');
export const PROPERTY_TOKEN_breaks = new InjectionToken('property-token-breaks');
export const PROPERTY_TOKEN_constantLines = new InjectionToken('property-token-constantLines');
export const PROPERTY_TOKEN_panes = new InjectionToken('property-token-panes');
export const PROPERTY_TOKEN_series = new InjectionToken('property-token-series');
export const PROPERTY_TOKEN_strips = new InjectionToken('property-token-strips');
export const PROPERTY_TOKEN_valueAxis = new InjectionToken('property-token-valueAxis');
export const PROPERTY_TOKEN_alerts = new InjectionToken('property-token-alerts');
export const PROPERTY_TOKEN_attachments = new InjectionToken('property-token-attachments');
export const PROPERTY_TOKEN_typingUsers = new InjectionToken('property-token-typingUsers');
export const PROPERTY_TOKEN_ranges = new InjectionToken('property-token-ranges');
export const PROPERTY_TOKEN_sortByGroupSummaryInfo = new InjectionToken('property-token-sortByGroupSummaryInfo');
export const PROPERTY_TOKEN_totalItems = new InjectionToken('property-token-totalItems');
export const PROPERTY_TOKEN_commands = new InjectionToken('property-token-commands');
export const PROPERTY_TOKEN_connectionPoints = new InjectionToken('property-token-connectionPoints');
export const PROPERTY_TOKEN_customShapes = new InjectionToken('property-token-customShapes');
export const PROPERTY_TOKEN_groups = new InjectionToken('property-token-groups');
export const PROPERTY_TOKEN_fileSelectionItems = new InjectionToken('property-token-fileSelectionItems');
export const PROPERTY_TOKEN_stripLines = new InjectionToken('property-token-stripLines');
export const PROPERTY_TOKEN_mentions = new InjectionToken('property-token-mentions');
export const PROPERTY_TOKEN_menuItems = new InjectionToken('property-token-menuItems');
export const PROPERTY_TOKEN_locations = new InjectionToken('property-token-locations');
export const PROPERTY_TOKEN_markers = new InjectionToken('property-token-markers');
export const PROPERTY_TOKEN_routes = new InjectionToken('property-token-routes');
export const PROPERTY_TOKEN_cols = new InjectionToken('property-token-cols');
export const PROPERTY_TOKEN_location = new InjectionToken('property-token-location');
export const PROPERTY_TOKEN_rows = new InjectionToken('property-token-rows');
export const PROPERTY_TOKEN_resources = new InjectionToken('property-token-resources');
export const PROPERTY_TOKEN_views = new InjectionToken('property-token-views');
export const PROPERTY_TOKEN_layers = new InjectionToken('property-token-layers');
export const PROPERTY_TOKEN_legends = new InjectionToken('property-token-legends');
export const PROPERTY_TOKEN_center = new InjectionToken('property-token-center');
export const PROPERTY_TOKEN_groupItems = new InjectionToken('property-token-groupItems');
