// devextreme-angular/esm2022/ui/color-box/nested/field-addons.mjs
/*!
 * devextreme-angular
 * Version: 25.2.3
 * Build date: Fri Dec 12 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */
/* tslint:disable:max-line-length */
import { Component, NgModule, Host, SkipSelf, Input } from '@angular/core';
import { DxIntegrationModule, NestedOptionHost, } from 'devextreme-angular/core';
import { NestedOption } from 'devextreme-angular/core';
import * as i0 from "@angular/core";
import * as i1 from "devextreme-angular/core";
export class DxoColorBoxFieldAddonsComponent extends NestedOption {
    get afterTemplate() {
        return this._getOption('afterTemplate');
    }
    set afterTemplate(value) {
        this._setOption('afterTemplate', value);
    }
    get beforeTemplate() {
        return this._getOption('beforeTemplate');
    }
    set beforeTemplate(value) {
        this._setOption('beforeTemplate', value);
    }
    get _optionPath() {
        return 'fieldAddons';
    }
    constructor(parentOptionHost, optionHost) {
        super();
        parentOptionHost.setNestedOption(this);
        optionHost.setHost(this, this._fullOptionPath.bind(this));
    }
    ngOnInit() {
        this._addRecreatedComponent();
    }
    ngOnDestroy() {
        this._addRemovedOption(this._getOptionPath());
    }
    /** @nocollapse */ static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxoColorBoxFieldAddonsComponent, deps: [{ token: i1.NestedOptionHost, host: true, skipSelf: true }, { token: i1.NestedOptionHost, host: true }], target: i0.ɵɵFactoryTarget.Component });
    /** @nocollapse */ static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.3.12", type: DxoColorBoxFieldAddonsComponent, isStandalone: true, selector: "dxo-color-box-field-addons", inputs: { afterTemplate: "afterTemplate", beforeTemplate: "beforeTemplate" }, providers: [NestedOptionHost], usesInheritance: true, ngImport: i0, template: '', isInline: true, styles: [""], dependencies: [{ kind: "ngmodule", type: DxIntegrationModule }] });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxoColorBoxFieldAddonsComponent, decorators: [{
            type: Component,
            args: [{ selector: 'dxo-color-box-field-addons', standalone: true, template: '', imports: [DxIntegrationModule], providers: [NestedOptionHost] }]
        }], ctorParameters: () => [{ type: i1.NestedOptionHost, decorators: [{
                    type: SkipSelf
                }, {
                    type: Host
                }] }, { type: i1.NestedOptionHost, decorators: [{
                    type: Host
                }] }], propDecorators: { afterTemplate: [{
                type: Input
            }], beforeTemplate: [{
                type: Input
            }] } });
export class DxoColorBoxFieldAddonsModule {
    /** @nocollapse */ static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxoColorBoxFieldAddonsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
    /** @nocollapse */ static ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.3.12", ngImport: i0, type: DxoColorBoxFieldAddonsModule, imports: [DxoColorBoxFieldAddonsComponent], exports: [DxoColorBoxFieldAddonsComponent] });
    /** @nocollapse */ static ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxoColorBoxFieldAddonsModule, imports: [DxoColorBoxFieldAddonsComponent] });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxoColorBoxFieldAddonsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        DxoColorBoxFieldAddonsComponent
                    ],
                    exports: [
                        DxoColorBoxFieldAddonsComponent
                    ],
                }]
        }] });
