// devextreme-angular/esm2022/ui/tree-list/nested/ai.mjs
/*!
 * devextreme-angular
 * Version: 25.2.3
 * Build date: Fri Dec 12 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */
/* tslint:disable:max-line-length */
import { Component, NgModule, Host, SkipSelf, Input } from '@angular/core';
import { AIIntegration } from 'devextreme/common/ai-integration';
import { DxIntegrationModule, NestedOptionHost, } from 'devextreme-angular/core';
import { NestedOption } from 'devextreme-angular/core';
import * as i0 from "@angular/core";
import * as i1 from "devextreme-angular/core";
export class DxoTreeListAIComponent extends NestedOption {
    get aiIntegration() {
        return this._getOption('aiIntegration');
    }
    set aiIntegration(value) {
        this._setOption('aiIntegration', value);
    }
    get editorOptions() {
        return this._getOption('editorOptions');
    }
    set editorOptions(value) {
        this._setOption('editorOptions', value);
    }
    get emptyText() {
        return this._getOption('emptyText');
    }
    set emptyText(value) {
        this._setOption('emptyText', value);
    }
    get mode() {
        return this._getOption('mode');
    }
    set mode(value) {
        this._setOption('mode', value);
    }
    get noDataText() {
        return this._getOption('noDataText');
    }
    set noDataText(value) {
        this._setOption('noDataText', value);
    }
    get popup() {
        return this._getOption('popup');
    }
    set popup(value) {
        this._setOption('popup', value);
    }
    get prompt() {
        return this._getOption('prompt');
    }
    set prompt(value) {
        this._setOption('prompt', value);
    }
    get showHeaderMenu() {
        return this._getOption('showHeaderMenu');
    }
    set showHeaderMenu(value) {
        this._setOption('showHeaderMenu', value);
    }
    get _optionPath() {
        return 'ai';
    }
    constructor(parentOptionHost, optionHost) {
        super();
        parentOptionHost.setNestedOption(this);
        optionHost.setHost(this, this._fullOptionPath.bind(this));
    }
    ngOnInit() {
        this._addRecreatedComponent();
    }
    ngOnDestroy() {
        this._addRemovedOption(this._getOptionPath());
    }
    /** @nocollapse */ static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxoTreeListAIComponent, deps: [{ token: i1.NestedOptionHost, host: true, skipSelf: true }, { token: i1.NestedOptionHost, host: true }], target: i0.ɵɵFactoryTarget.Component });
    /** @nocollapse */ static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.3.12", type: DxoTreeListAIComponent, isStandalone: true, selector: "dxo-tree-list-ai", inputs: { aiIntegration: "aiIntegration", editorOptions: "editorOptions", emptyText: "emptyText", mode: "mode", noDataText: "noDataText", popup: "popup", prompt: "prompt", showHeaderMenu: "showHeaderMenu" }, providers: [NestedOptionHost], usesInheritance: true, ngImport: i0, template: '', isInline: true, styles: [""], dependencies: [{ kind: "ngmodule", type: DxIntegrationModule }] });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxoTreeListAIComponent, decorators: [{
            type: Component,
            args: [{ selector: 'dxo-tree-list-ai', standalone: true, template: '', imports: [DxIntegrationModule], providers: [NestedOptionHost] }]
        }], ctorParameters: () => [{ type: i1.NestedOptionHost, decorators: [{
                    type: SkipSelf
                }, {
                    type: Host
                }] }, { type: i1.NestedOptionHost, decorators: [{
                    type: Host
                }] }], propDecorators: { aiIntegration: [{
                type: Input
            }], editorOptions: [{
                type: Input
            }], emptyText: [{
                type: Input
            }], mode: [{
                type: Input
            }], noDataText: [{
                type: Input
            }], popup: [{
                type: Input
            }], prompt: [{
                type: Input
            }], showHeaderMenu: [{
                type: Input
            }] } });
export class DxoTreeListAIModule {
    /** @nocollapse */ static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxoTreeListAIModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
    /** @nocollapse */ static ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.3.12", ngImport: i0, type: DxoTreeListAIModule, imports: [DxoTreeListAIComponent], exports: [DxoTreeListAIComponent] });
    /** @nocollapse */ static ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxoTreeListAIModule, imports: [DxoTreeListAIComponent] });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxoTreeListAIModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        DxoTreeListAIComponent
                    ],
                    exports: [
                        DxoTreeListAIComponent
                    ],
                }]
        }] });
