// devextreme-angular/fesm2022/devextreme-angular-ui-speech-to-text-nested.mjs
import * as i0 from '@angular/core';
import { Component, SkipSelf, Host, Input, NgModule } from '@angular/core';
import * as i1 from 'devextreme-angular/core';
import { NestedOption, NestedOptionHost, DxIntegrationModule } from 'devextreme-angular/core';

/*!
 * devextreme-angular
 * Version: 25.2.3
 * Build date: Fri Dec 12 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */
/* tslint:disable:max-line-length */
class DxoSpeechToTextCustomSpeechRecognizerComponent extends NestedOption {
    get enabled() {
        return this._getOption('enabled');
    }
    set enabled(value) {
        this._setOption('enabled', value);
    }
    get isListening() {
        return this._getOption('isListening');
    }
    set isListening(value) {
        this._setOption('isListening', value);
    }
    get _optionPath() {
        return 'customSpeechRecognizer';
    }
    constructor(parentOptionHost, optionHost) {
        super();
        parentOptionHost.setNestedOption(this);
        optionHost.setHost(this, this._fullOptionPath.bind(this));
    }
    ngOnInit() {
        this._addRecreatedComponent();
    }
    ngOnDestroy() {
        this._addRemovedOption(this._getOptionPath());
    }
    /** @nocollapse */ static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxoSpeechToTextCustomSpeechRecognizerComponent, deps: [{ token: i1.NestedOptionHost, host: true, skipSelf: true }, { token: i1.NestedOptionHost, host: true }], target: i0.ɵɵFactoryTarget.Component });
    /** @nocollapse */ static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.3.12", type: DxoSpeechToTextCustomSpeechRecognizerComponent, isStandalone: true, selector: "dxo-speech-to-text-custom-speech-recognizer", inputs: { enabled: "enabled", isListening: "isListening" }, providers: [NestedOptionHost], usesInheritance: true, ngImport: i0, template: '', isInline: true, styles: [""], dependencies: [{ kind: "ngmodule", type: DxIntegrationModule }] });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxoSpeechToTextCustomSpeechRecognizerComponent, decorators: [{
            type: Component,
            args: [{ selector: 'dxo-speech-to-text-custom-speech-recognizer', standalone: true, template: '', imports: [DxIntegrationModule], providers: [NestedOptionHost] }]
        }], ctorParameters: () => [{ type: i1.NestedOptionHost, decorators: [{
                    type: SkipSelf
                }, {
                    type: Host
                }] }, { type: i1.NestedOptionHost, decorators: [{
                    type: Host
                }] }], propDecorators: { enabled: [{
                type: Input
            }], isListening: [{
                type: Input
            }] } });
class DxoSpeechToTextCustomSpeechRecognizerModule {
    /** @nocollapse */ static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxoSpeechToTextCustomSpeechRecognizerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
    /** @nocollapse */ static ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.3.12", ngImport: i0, type: DxoSpeechToTextCustomSpeechRecognizerModule, imports: [DxoSpeechToTextCustomSpeechRecognizerComponent], exports: [DxoSpeechToTextCustomSpeechRecognizerComponent] });
    /** @nocollapse */ static ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxoSpeechToTextCustomSpeechRecognizerModule, imports: [DxoSpeechToTextCustomSpeechRecognizerComponent] });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxoSpeechToTextCustomSpeechRecognizerModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        DxoSpeechToTextCustomSpeechRecognizerComponent
                    ],
                    exports: [
                        DxoSpeechToTextCustomSpeechRecognizerComponent
                    ],
                }]
        }] });

/*!
 * devextreme-angular
 * Version: 25.2.3
 * Build date: Fri Dec 12 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */
/* tslint:disable:max-line-length */
class DxoSpeechToTextSpeechRecognitionConfigComponent extends NestedOption {
    get continuous() {
        return this._getOption('continuous');
    }
    set continuous(value) {
        this._setOption('continuous', value);
    }
    get grammars() {
        return this._getOption('grammars');
    }
    set grammars(value) {
        this._setOption('grammars', value);
    }
    get interimResults() {
        return this._getOption('interimResults');
    }
    set interimResults(value) {
        this._setOption('interimResults', value);
    }
    get lang() {
        return this._getOption('lang');
    }
    set lang(value) {
        this._setOption('lang', value);
    }
    get maxAlternatives() {
        return this._getOption('maxAlternatives');
    }
    set maxAlternatives(value) {
        this._setOption('maxAlternatives', value);
    }
    get _optionPath() {
        return 'speechRecognitionConfig';
    }
    constructor(parentOptionHost, optionHost) {
        super();
        parentOptionHost.setNestedOption(this);
        optionHost.setHost(this, this._fullOptionPath.bind(this));
    }
    ngOnInit() {
        this._addRecreatedComponent();
    }
    ngOnDestroy() {
        this._addRemovedOption(this._getOptionPath());
    }
    /** @nocollapse */ static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxoSpeechToTextSpeechRecognitionConfigComponent, deps: [{ token: i1.NestedOptionHost, host: true, skipSelf: true }, { token: i1.NestedOptionHost, host: true }], target: i0.ɵɵFactoryTarget.Component });
    /** @nocollapse */ static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.3.12", type: DxoSpeechToTextSpeechRecognitionConfigComponent, isStandalone: true, selector: "dxo-speech-to-text-speech-recognition-config", inputs: { continuous: "continuous", grammars: "grammars", interimResults: "interimResults", lang: "lang", maxAlternatives: "maxAlternatives" }, providers: [NestedOptionHost], usesInheritance: true, ngImport: i0, template: '', isInline: true, styles: [""], dependencies: [{ kind: "ngmodule", type: DxIntegrationModule }] });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxoSpeechToTextSpeechRecognitionConfigComponent, decorators: [{
            type: Component,
            args: [{ selector: 'dxo-speech-to-text-speech-recognition-config', standalone: true, template: '', imports: [DxIntegrationModule], providers: [NestedOptionHost] }]
        }], ctorParameters: () => [{ type: i1.NestedOptionHost, decorators: [{
                    type: SkipSelf
                }, {
                    type: Host
                }] }, { type: i1.NestedOptionHost, decorators: [{
                    type: Host
                }] }], propDecorators: { continuous: [{
                type: Input
            }], grammars: [{
                type: Input
            }], interimResults: [{
                type: Input
            }], lang: [{
                type: Input
            }], maxAlternatives: [{
                type: Input
            }] } });
class DxoSpeechToTextSpeechRecognitionConfigModule {
    /** @nocollapse */ static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxoSpeechToTextSpeechRecognitionConfigModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
    /** @nocollapse */ static ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.3.12", ngImport: i0, type: DxoSpeechToTextSpeechRecognitionConfigModule, imports: [DxoSpeechToTextSpeechRecognitionConfigComponent], exports: [DxoSpeechToTextSpeechRecognitionConfigComponent] });
    /** @nocollapse */ static ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxoSpeechToTextSpeechRecognitionConfigModule, imports: [DxoSpeechToTextSpeechRecognitionConfigComponent] });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxoSpeechToTextSpeechRecognitionConfigModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        DxoSpeechToTextSpeechRecognitionConfigComponent
                    ],
                    exports: [
                        DxoSpeechToTextSpeechRecognitionConfigComponent
                    ],
                }]
        }] });

/*!
 * devextreme-angular
 * Version: 25.2.3
 * Build date: Fri Dec 12 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */

/**
 * Generated bundle index. Do not edit.
 */

export { DxoSpeechToTextCustomSpeechRecognizerComponent, DxoSpeechToTextCustomSpeechRecognizerModule, DxoSpeechToTextSpeechRecognitionConfigComponent, DxoSpeechToTextSpeechRecognitionConfigModule };
