// devextreme-angular/esm2022/ui/speech-to-text/index.mjs
/*!
 * devextreme-angular
 * Version: 25.2.3
 * Build date: Fri Dec 12 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */
/* tslint:disable:max-line-length */
import { TransferState, Component, NgModule, ElementRef, NgZone, PLATFORM_ID, Inject, Input, Output, EventEmitter } from '@angular/core';
import DxSpeechToText from 'devextreme/ui/speech_to_text';
import { DxComponent, DxTemplateHost, DxIntegrationModule, DxTemplateModule, NestedOptionHost, WatcherHelper, } from 'devextreme-angular/core';
import { DxoSpeechToTextCustomSpeechRecognizerModule } from 'devextreme-angular/ui/speech-to-text/nested';
import { DxoSpeechToTextSpeechRecognitionConfigModule } from 'devextreme-angular/ui/speech-to-text/nested';
import * as i0 from "@angular/core";
import * as i1 from "devextreme-angular/core";
export class DxSpeechToTextComponent extends DxComponent {
    instance = null;
    get accessKey() {
        return this._getOption('accessKey');
    }
    set accessKey(value) {
        this._setOption('accessKey', value);
    }
    get activeStateEnabled() {
        return this._getOption('activeStateEnabled');
    }
    set activeStateEnabled(value) {
        this._setOption('activeStateEnabled', value);
    }
    get customSpeechRecognizer() {
        return this._getOption('customSpeechRecognizer');
    }
    set customSpeechRecognizer(value) {
        this._setOption('customSpeechRecognizer', value);
    }
    get disabled() {
        return this._getOption('disabled');
    }
    set disabled(value) {
        this._setOption('disabled', value);
    }
    get elementAttr() {
        return this._getOption('elementAttr');
    }
    set elementAttr(value) {
        this._setOption('elementAttr', value);
    }
    get focusStateEnabled() {
        return this._getOption('focusStateEnabled');
    }
    set focusStateEnabled(value) {
        this._setOption('focusStateEnabled', value);
    }
    get height() {
        return this._getOption('height');
    }
    set height(value) {
        this._setOption('height', value);
    }
    get hint() {
        return this._getOption('hint');
    }
    set hint(value) {
        this._setOption('hint', value);
    }
    get hoverStateEnabled() {
        return this._getOption('hoverStateEnabled');
    }
    set hoverStateEnabled(value) {
        this._setOption('hoverStateEnabled', value);
    }
    get rtlEnabled() {
        return this._getOption('rtlEnabled');
    }
    set rtlEnabled(value) {
        this._setOption('rtlEnabled', value);
    }
    get speechRecognitionConfig() {
        return this._getOption('speechRecognitionConfig');
    }
    set speechRecognitionConfig(value) {
        this._setOption('speechRecognitionConfig', value);
    }
    get startIcon() {
        return this._getOption('startIcon');
    }
    set startIcon(value) {
        this._setOption('startIcon', value);
    }
    get startText() {
        return this._getOption('startText');
    }
    set startText(value) {
        this._setOption('startText', value);
    }
    get stopIcon() {
        return this._getOption('stopIcon');
    }
    set stopIcon(value) {
        this._setOption('stopIcon', value);
    }
    get stopText() {
        return this._getOption('stopText');
    }
    set stopText(value) {
        this._setOption('stopText', value);
    }
    get stylingMode() {
        return this._getOption('stylingMode');
    }
    set stylingMode(value) {
        this._setOption('stylingMode', value);
    }
    get tabIndex() {
        return this._getOption('tabIndex');
    }
    set tabIndex(value) {
        this._setOption('tabIndex', value);
    }
    get type() {
        return this._getOption('type');
    }
    set type(value) {
        this._setOption('type', value);
    }
    get visible() {
        return this._getOption('visible');
    }
    set visible(value) {
        this._setOption('visible', value);
    }
    get width() {
        return this._getOption('width');
    }
    set width(value) {
        this._setOption('width', value);
    }
    /**
    
     * 
    
    
     */
    onContentReady;
    /**
    
     * 
    
    
     */
    onDisposing;
    /**
    
     * 
    
    
     */
    onEnd;
    /**
    
     * 
    
    
     */
    onError;
    /**
    
     * 
    
    
     */
    onInitialized;
    /**
    
     * 
    
    
     */
    onOptionChanged;
    /**
    
     * 
    
    
     */
    onResult;
    /**
    
     * 
    
    
     */
    onStartClick;
    /**
    
     * 
    
    
     */
    onStopClick;
    /**
    
     * This member supports the internal infrastructure and is not intended to be used directly from your code.
    
     */
    accessKeyChange;
    /**
    
     * This member supports the internal infrastructure and is not intended to be used directly from your code.
    
     */
    activeStateEnabledChange;
    /**
    
     * This member supports the internal infrastructure and is not intended to be used directly from your code.
    
     */
    customSpeechRecognizerChange;
    /**
    
     * This member supports the internal infrastructure and is not intended to be used directly from your code.
    
     */
    disabledChange;
    /**
    
     * This member supports the internal infrastructure and is not intended to be used directly from your code.
    
     */
    elementAttrChange;
    /**
    
     * This member supports the internal infrastructure and is not intended to be used directly from your code.
    
     */
    focusStateEnabledChange;
    /**
    
     * This member supports the internal infrastructure and is not intended to be used directly from your code.
    
     */
    heightChange;
    /**
    
     * This member supports the internal infrastructure and is not intended to be used directly from your code.
    
     */
    hintChange;
    /**
    
     * This member supports the internal infrastructure and is not intended to be used directly from your code.
    
     */
    hoverStateEnabledChange;
    /**
    
     * This member supports the internal infrastructure and is not intended to be used directly from your code.
    
     */
    rtlEnabledChange;
    /**
    
     * This member supports the internal infrastructure and is not intended to be used directly from your code.
    
     */
    speechRecognitionConfigChange;
    /**
    
     * This member supports the internal infrastructure and is not intended to be used directly from your code.
    
     */
    startIconChange;
    /**
    
     * This member supports the internal infrastructure and is not intended to be used directly from your code.
    
     */
    startTextChange;
    /**
    
     * This member supports the internal infrastructure and is not intended to be used directly from your code.
    
     */
    stopIconChange;
    /**
    
     * This member supports the internal infrastructure and is not intended to be used directly from your code.
    
     */
    stopTextChange;
    /**
    
     * This member supports the internal infrastructure and is not intended to be used directly from your code.
    
     */
    stylingModeChange;
    /**
    
     * This member supports the internal infrastructure and is not intended to be used directly from your code.
    
     */
    tabIndexChange;
    /**
    
     * This member supports the internal infrastructure and is not intended to be used directly from your code.
    
     */
    typeChange;
    /**
    
     * This member supports the internal infrastructure and is not intended to be used directly from your code.
    
     */
    visibleChange;
    /**
    
     * This member supports the internal infrastructure and is not intended to be used directly from your code.
    
     */
    widthChange;
    constructor(elementRef, ngZone, templateHost, _watcherHelper, optionHost, transferState, platformId) {
        super(elementRef, ngZone, templateHost, _watcherHelper, transferState, platformId);
        this._createEventEmitters([
            { subscribe: 'contentReady', emit: 'onContentReady' },
            { subscribe: 'disposing', emit: 'onDisposing' },
            { subscribe: 'end', emit: 'onEnd' },
            { subscribe: 'error', emit: 'onError' },
            { subscribe: 'initialized', emit: 'onInitialized' },
            { subscribe: 'optionChanged', emit: 'onOptionChanged' },
            { subscribe: 'result', emit: 'onResult' },
            { subscribe: 'startClick', emit: 'onStartClick' },
            { subscribe: 'stopClick', emit: 'onStopClick' },
            { emit: 'accessKeyChange' },
            { emit: 'activeStateEnabledChange' },
            { emit: 'customSpeechRecognizerChange' },
            { emit: 'disabledChange' },
            { emit: 'elementAttrChange' },
            { emit: 'focusStateEnabledChange' },
            { emit: 'heightChange' },
            { emit: 'hintChange' },
            { emit: 'hoverStateEnabledChange' },
            { emit: 'rtlEnabledChange' },
            { emit: 'speechRecognitionConfigChange' },
            { emit: 'startIconChange' },
            { emit: 'startTextChange' },
            { emit: 'stopIconChange' },
            { emit: 'stopTextChange' },
            { emit: 'stylingModeChange' },
            { emit: 'tabIndexChange' },
            { emit: 'typeChange' },
            { emit: 'visibleChange' },
            { emit: 'widthChange' }
        ]);
        optionHost.setHost(this);
    }
    _createInstance(element, options) {
        return new DxSpeechToText(element, options);
    }
    ngOnDestroy() {
        this._destroyWidget();
    }
    /** @nocollapse */ static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxSpeechToTextComponent, deps: [{ token: i0.ElementRef }, { token: i0.NgZone }, { token: i1.DxTemplateHost }, { token: i1.WatcherHelper }, { token: i1.NestedOptionHost }, { token: i0.TransferState }, { token: PLATFORM_ID }], target: i0.ɵɵFactoryTarget.Component });
    /** @nocollapse */ static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.3.12", type: DxSpeechToTextComponent, isStandalone: true, selector: "dx-speech-to-text", inputs: { accessKey: "accessKey", activeStateEnabled: "activeStateEnabled", customSpeechRecognizer: "customSpeechRecognizer", disabled: "disabled", elementAttr: "elementAttr", focusStateEnabled: "focusStateEnabled", height: "height", hint: "hint", hoverStateEnabled: "hoverStateEnabled", rtlEnabled: "rtlEnabled", speechRecognitionConfig: "speechRecognitionConfig", startIcon: "startIcon", startText: "startText", stopIcon: "stopIcon", stopText: "stopText", stylingMode: "stylingMode", tabIndex: "tabIndex", type: "type", visible: "visible", width: "width" }, outputs: { onContentReady: "onContentReady", onDisposing: "onDisposing", onEnd: "onEnd", onError: "onError", onInitialized: "onInitialized", onOptionChanged: "onOptionChanged", onResult: "onResult", onStartClick: "onStartClick", onStopClick: "onStopClick", accessKeyChange: "accessKeyChange", activeStateEnabledChange: "activeStateEnabledChange", customSpeechRecognizerChange: "customSpeechRecognizerChange", disabledChange: "disabledChange", elementAttrChange: "elementAttrChange", focusStateEnabledChange: "focusStateEnabledChange", heightChange: "heightChange", hintChange: "hintChange", hoverStateEnabledChange: "hoverStateEnabledChange", rtlEnabledChange: "rtlEnabledChange", speechRecognitionConfigChange: "speechRecognitionConfigChange", startIconChange: "startIconChange", startTextChange: "startTextChange", stopIconChange: "stopIconChange", stopTextChange: "stopTextChange", stylingModeChange: "stylingModeChange", tabIndexChange: "tabIndexChange", typeChange: "typeChange", visibleChange: "visibleChange", widthChange: "widthChange" }, host: { attributes: { "ngSkipHydration": "true" } }, providers: [
            DxTemplateHost,
            WatcherHelper,
            NestedOptionHost
        ], usesInheritance: true, ngImport: i0, template: '', isInline: true, dependencies: [{ kind: "ngmodule", type: DxIntegrationModule }] });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxSpeechToTextComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'dx-speech-to-text',
                    standalone: true,
                    template: '',
                    host: { ngSkipHydration: 'true' },
                    imports: [DxIntegrationModule],
                    providers: [
                        DxTemplateHost,
                        WatcherHelper,
                        NestedOptionHost
                    ]
                }]
        }], ctorParameters: () => [{ type: i0.ElementRef }, { type: i0.NgZone }, { type: i1.DxTemplateHost }, { type: i1.WatcherHelper }, { type: i1.NestedOptionHost }, { type: i0.TransferState }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [PLATFORM_ID]
                }] }], propDecorators: { accessKey: [{
                type: Input
            }], activeStateEnabled: [{
                type: Input
            }], customSpeechRecognizer: [{
                type: Input
            }], disabled: [{
                type: Input
            }], elementAttr: [{
                type: Input
            }], focusStateEnabled: [{
                type: Input
            }], height: [{
                type: Input
            }], hint: [{
                type: Input
            }], hoverStateEnabled: [{
                type: Input
            }], rtlEnabled: [{
                type: Input
            }], speechRecognitionConfig: [{
                type: Input
            }], startIcon: [{
                type: Input
            }], startText: [{
                type: Input
            }], stopIcon: [{
                type: Input
            }], stopText: [{
                type: Input
            }], stylingMode: [{
                type: Input
            }], tabIndex: [{
                type: Input
            }], type: [{
                type: Input
            }], visible: [{
                type: Input
            }], width: [{
                type: Input
            }], onContentReady: [{
                type: Output
            }], onDisposing: [{
                type: Output
            }], onEnd: [{
                type: Output
            }], onError: [{
                type: Output
            }], onInitialized: [{
                type: Output
            }], onOptionChanged: [{
                type: Output
            }], onResult: [{
                type: Output
            }], onStartClick: [{
                type: Output
            }], onStopClick: [{
                type: Output
            }], accessKeyChange: [{
                type: Output
            }], activeStateEnabledChange: [{
                type: Output
            }], customSpeechRecognizerChange: [{
                type: Output
            }], disabledChange: [{
                type: Output
            }], elementAttrChange: [{
                type: Output
            }], focusStateEnabledChange: [{
                type: Output
            }], heightChange: [{
                type: Output
            }], hintChange: [{
                type: Output
            }], hoverStateEnabledChange: [{
                type: Output
            }], rtlEnabledChange: [{
                type: Output
            }], speechRecognitionConfigChange: [{
                type: Output
            }], startIconChange: [{
                type: Output
            }], startTextChange: [{
                type: Output
            }], stopIconChange: [{
                type: Output
            }], stopTextChange: [{
                type: Output
            }], stylingModeChange: [{
                type: Output
            }], tabIndexChange: [{
                type: Output
            }], typeChange: [{
                type: Output
            }], visibleChange: [{
                type: Output
            }], widthChange: [{
                type: Output
            }] } });
export class DxSpeechToTextModule {
    /** @nocollapse */ static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxSpeechToTextModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
    /** @nocollapse */ static ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.3.12", ngImport: i0, type: DxSpeechToTextModule, imports: [DxSpeechToTextComponent, DxoSpeechToTextCustomSpeechRecognizerModule,
            DxoSpeechToTextSpeechRecognitionConfigModule,
            DxIntegrationModule,
            DxTemplateModule], exports: [DxSpeechToTextComponent, DxoSpeechToTextCustomSpeechRecognizerModule,
            DxoSpeechToTextSpeechRecognitionConfigModule,
            DxTemplateModule] });
    /** @nocollapse */ static ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxSpeechToTextModule, imports: [DxSpeechToTextComponent,
            DxoSpeechToTextCustomSpeechRecognizerModule,
            DxoSpeechToTextSpeechRecognitionConfigModule,
            DxIntegrationModule,
            DxTemplateModule, DxoSpeechToTextCustomSpeechRecognizerModule,
            DxoSpeechToTextSpeechRecognitionConfigModule,
            DxTemplateModule] });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxSpeechToTextModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        DxSpeechToTextComponent,
                        DxoSpeechToTextCustomSpeechRecognizerModule,
                        DxoSpeechToTextSpeechRecognitionConfigModule,
                        DxIntegrationModule,
                        DxTemplateModule
                    ],
                    exports: [
                        DxSpeechToTextComponent,
                        DxoSpeechToTextCustomSpeechRecognizerModule,
                        DxoSpeechToTextSpeechRecognitionConfigModule,
                        DxTemplateModule
                    ]
                }]
        }] });
export * from 'devextreme-angular/ui/speech-to-text/nested';
